### DISCUSSION TITLE
[Soroban Settings upgrade] Reduce the rent costs and volatility

### DISCUSSION THREAD

**Discussion by dmkozh** [stellar/stellar-protocol#1488] [Core Advancement Proposals]
This is a proposal to update Soroban settings with a goal of improving the usability and reducing the overall costs for maintaining and interacting with the Smart Contracts. The proposal also bundles a per-transaction read limit increase as to enable support of more complex protocols.

The following pain points motivate the proposal:

- Write fees and consequently cost of rent depend on the size of the ledger state ('bucket list size'), and currently this dependency is way too steep, leading to possible 2-3x write fee changes within a few days.
- The absolute value of the rent fees is pretty high.

While the storage fees have to be reasonably high as to prevent explosive ledger state growth, we think that we can improve the UX without changing the protocol, while still maintaining automatic fee-based protection of the network from too quick growth. Here is what is proposed:

- Make the write fee scaling very flat: scale from **8510** to **12116** stroops/KB while bucket list size growth from 0 to 12.5 GB in size. This means that the Pubnet fees will fluctuate slowly between ~11500 and ~12000 stroops/KB given the current usage patterns
- Increase the scaling factor past the size threshold (12.5 GB) to **5000** as to make sure the fees still grow quickly if the bucket list size is too large
- Increase the rent period 1.5x, thus reducing the effective rent costs 1.5x for all the ledger entries
- Increase the read bytes limit per transaction to 200 KB. The total per-ledger limit on the amount of read bytes stays the same (500 KB), so there are no performance-related concerns. 

The exact proposed setting changes can be found in https://github.com/stellar/stellar-core/pull/4335. The Testnet changes follow this pattern, except it has a bit different write fee scaling as to account for much smaller state size and get the effective fees close to those on Pubnet.

Note, that the rent-related changes are meant as a short term improvement for the general issues with the rent costs on Soroban. However, more holistic changes would require a protocol change and thus will only be possible to make with the next protocol release.

**Comment by namankumar** [comment]
Related discord thread: https://discord.com/channels/897514728459468821/1246171455692214323
